### PR TITLE
feat: [#493] optimize session

### PR DESCRIPTION
**Pull request by kkumar-gcc**
## 📑 Description

RelatedTo https://github.com/goravel/goravel/issues/493

## Summary by CodeRabbit

- **New Features**
	- Enhanced session management with improved error handling and validation.
	- Introduced a new method to set the session driver externally.

- **Bug Fixes**
	- Improved error reporting for unset session drivers.
	- Added error handling in session creation to prevent unhandled exceptions.

- **Tests**
	- Expanded test cases to ensure proper handling of session driver errors and session integrity.

- **Documentation**
	- Added descriptive error messages for better clarity in session management.

## ✅ Checks

- [ ] Added test cases for my code

### CODE REVIEW

**Comment by devhaozi** [comment]
I have another question, what is the use of `_token` in session?

**Comment by kkumar-gcc** [comment]
> I have another question, what is the use of `_token` in session? It seems to just set but not used.

I initially started implementing session-based CSRF verification middleware but later dropped the idea. The _token is typically used for CSRF protection. (and result was session module)

File: session/session.go
Comment (hwbrzzl) [inline session/session.go:1]: Please update test cases based on the latest logic.

**Review by hwbrzzl — COMMENTED** [review]
Clearer 👍 Only update test cases.

File: session/session.go
Comment (kkumar-gcc) [reply session/session.go:1]: image

Should we return error here(`Start`) instead of printing?

Comment (hwbrzzl) [reply session/session.go:1]: Where print?

File: session/errors.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package session
+
+import "errors"
+
+var (
+	ErrDriverNotSet = errors.New("session driver is not set")
+	ErrDriverIsNil  = errors.New("session driver cannot be nil")
```
Comment (hwbrzzl) [inline session/errors.go:7]: They seem to be the same.

File: session/session.go
Comment (kkumar-gcc) [reply session/session.go:1]: in `readFromHandler`

Comment (hwbrzzl) [reply session/session.go:1]: `readFromHandler` and `loadSession` can return an error, and the error can be print in `Start`.

**Review by devhaozi — APPROVED** [review]
LGTM

**Review by hwbrzzl — APPROVED** [review]
Great PR 👍

**Pull request merged**